pkg/mongo/user.go

```diff
@@ -23,13 +28,20 @@ func (s *Service) AddUser(u *User) error {
 	return nil
 }
 
-/*
-func (s *Service) ReadUser() error {
-	u := &User{}
-	err := s.db.C("users").Find(u.Name)
-	if err != nil {
-		log.Println("incorrect collection")
+// ReadUser retrieves an information from the database and compares it with a request.
+func (s *Service) ReadUser(username string) (User, int, error) {
+	var result User
+	filter := bson.D{{"name", username}}
+	err := s.db.Collection("users").FindOne(context.Background(), filter).Decode(&result)
+	fmt.Println(result.Name, result.Hash)
+	switch err != nil {
+	case err == errors.New("mongo: no documents in result"):
+		return result, http.StatusUnauthorized, err
+	case err == errors.New("Registry cannot be nil"):
+		return result, http.StatusUnauthorized, err
+	case err != nil:
+		return result, http.StatusInternalServerError, err
```

Comment:
Why mongo package knows anything about HTTP status codes? this is a bad design. db package should operate on custom errors only and do not rely on other top-level packages